botorch/optim/optimize_mixed.py

```diff
@@ -552,6 +648,7 @@ def optimize_acqf_mixed_alternating(
     acq_function: AcquisitionFunction,
     bounds: Tensor,
     discrete_dims: list[int],
```

Comment:
would we want to make `discrete_dims` `None`able as well if there are only categorical dims?